h/static/scripts/group-forms/hooks/use-group-annotations.tsx

```diff
@@ -42,6 +52,29 @@ export function useGroupAnnotations({
   const [totalAnnotations, setTotalAnnotations] = useState(0);
   const [loading, setLoading] = useState(false);
   const [error, setError] = useState<string>();
+  const updateAnnotationStatus = useCallback(
+    (annotationId: string, moderationStatus: ModerationStatus) => {
+      if (filterStatus === undefined) {
+        // If no filter status is set, simply change the annotation's status to
+        // the new one
+        setAnnotations(prev =>
+          prev?.reduce<APIAnnotationData[]>((annos, currentAnno) => {
+            if (currentAnno.id === annotationId) {
+              currentAnno.moderation_status = moderationStatus;
```

Comment:
This mutates items in the list, which might cause unexpected behavior code that has a reference to the annotation from the old list. The "correct" thing to do would be to create a shallow copy and update the moderation status.

Reply:
Right!